### PR TITLE
Evagelos/ci 177

### PR DESCRIPTION
**Pull request by ev-agelos**
This PR is re-based on the #44 PR

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -5,61 +5,78 @@
...
+            steps {
+                slackNotify("STARTED")
+                sh 'rm -rf repo-state.txt venv'
+                checkout scm
+                sh "spynl/cli/ops/prepare-stage.sh -u $scm_urls -r $revision -f $fallbackrevision -m"
+            }
+            post {
+                always {
+                    archive 'repo-state.txt'                              // for history
+                    stash name:'repo-state', includes:'**/repo-state.txt' // for using it again later in this build
```
Comment (Kareeeeem) [inline Jenkinsfile]: remove

File: Jenkinsfile [hunk 2]
```diff
@@ -5,61 +5,78 @@
...
-      checkout scm
-      unstash name:'repo-state'
-      sh "spynl/cli/ops/prepare-stage.sh -u $scm_urls -r $revision -f $fallbackrevision"
-      sh "source venv/bin/activate && spynl ops.deploy --buildnr ${env.BUILD_NUMBER} --task $task"
-      archive 'spynl/cli/ops/docker/docker.build.log'  // for debugging
-    }
+        stage('Deploy') {  // Build Docker Image and deploy to dev or production ECR
+            steps {
+                checkout scm
+                unstash name:'repo-state'
```
Comment (Kareeeeem) [inline Jenkinsfile]: remove 2 lines

File: Jenkinsfile [hunk 3]
```diff
@@ -5,61 +5,78 @@
...
-  try {
+pipeline {
+    agent any
+    stages {
 
-    slackNotify("STARTED")
+        stage('Prepare build') {  // Cleanup, installations etc
+            steps {
+                slackNotify("STARTED")
+                sh 'rm -rf repo-state.txt venv'
```
Comment (Kareeeeem) [inline Jenkinsfile]: configure this as part of the pipeline behavior

**Review by Kareeeeem — APPROVED** [review]

**Review by Kareeeeem — APPROVED** [review]

File: Jenkinsfile [hunk 4]
```diff
@@ -5,61 +5,73 @@
...
-      //coverage 'venv/**/coverage.xml'  // Check this ticket: https://issues.jenkins-ci.org/browse/JENKINS-30700
-    }
+        stage('Unit Tests') {
+            steps {
+                sh "source venv/bin/activate && pip install pytest-cov && spynl dev.test --reports"
+            }
+            post {
+                always {
+                    junit 'venv/**/pytests.xml'
+                    //coverage 'venv/**/coverage.xml'  // Check this ticket: https://issues.jenkins-ci.org/browse/JENKINS-30700
```
Comment (marcin-wosinek) [inline Jenkinsfile:32]: we publish converage raport in html on v6:
https://jenkins.swcloud.nl/job/softwear6/job/build/Unit_test_coverage/
it's done like this:
https://bitbucket.org/softwear/softwear6/commits/940be7e8ccb7cee593d7475c46f38432b194b788

**Review by marcin-wosinek — APPROVED** [review]
looks good.

**Review by o-m-n-i — APPROVED** [review]
Looks good to me

File: Jenkinsfile (hunk 4 above)
Comment (ev-agelos) [reply Jenkinsfile:32]: I will come back to this as I dont think the little graph works (that tracks tests) for spynl.
Ill try to fix it in the next iteration, thanks.

**Pull request merged**